beetmoverscript/src/beetmoverscript/script.py

```diff
@@ -167,6 +167,37 @@ async def push_to_partner(context):
     add_checksums_to_artifacts(context)
 
 
+# copy_artifacts {{{1
+async def copy_artifacts(context):
+    """A un-opinionated action to copy artifacts from one location to another. It WONT overwrite existing artifacts in the destination."""
+    context.artifacts_to_beetmove = {}
+    product = context.task["payload"]["product"]
+    context.bucket_name = get_bucket_name(context, product, "aws")
+    copy_from = context.task["payload"]["copy_from"]
+    copy_to = context.task["payload"]["copy_to"]
+    exclude_files = context.task["payload"].get("exclude_files")
+
+    creds = get_creds(context)
+    s3_resource = boto3.resource("s3", aws_access_key_id=creds["id"], aws_secret_access_key=creds["key"])
+
+    from_keys = list_bucket_objects(context, s3_resource, copy_from)
+    to_keys = list_bucket_objects(context, s3_resource, copy_to)
+
+    if not from_keys:
+        raise ScriptWorkerTaskException("No artifacts to copy from {} so there is no reason to continue.".format(copy_from))
+
+    if to_keys:
```

Comment:
<s>It may be nice to know if we have something different in `copy_from` than exists in `copy_to`, or if it's the same file.

We could maybe move [copy_key](https://github.com/mozilla-releng/scriptworker-scripts/blob/8fea664a6d732d66767e16c8ce5c984a70db7e23/beetmoverscript/src/beetmoverscript/script.py#L249-L261) into a top level helper function, and use that.</s>

N/m, you're using `copy_beets` below :)